### PR TITLE
shim-sev: enable post-mortem stack traceback parsing

### PR DESCRIPTION
**Pull request by haraldh**
This patch series enables post-mortem stack traceback parsing as described in the `README-DEBUG.md`.

This helps while developing the shim, and even debug failures in production.

The stack trace makes use of stack frame pointers and needs the shim binary compiled with `-C force-frame-pointers=yes` or in case of a gcc payload `-fno-omit-frame-pointer` (payload debug normally not needed). While this blows up the size a little bit, I think the value exceeds the additional amount of bytes.

In case of a `panic()` the stack trace is printed on `stderr` of the `enarx-keepldr`. A future patch can introduce a special `panic` syscall where the `enarx-keepldr` can decide how to output the trace.

In case the shim can only do a hard VM shutdown, the traceback is provided in the registers.

To provide debug info for the `release` profile, `debug=1` is set in the profile. The debug information is removed with `strip` before including the shim in the loader, so the post-mortem trace back parser script has to work with the unstripped binary.

### CODE REVIEW

**Review by connorkuehl — APPROVED** [review]

File: internal/shim-sev/src/asm.rs [hunk 1]
```diff
@@ -11,15 +20,54 @@
...
+    in("r10") frames[10],
+    in("r11") frames[11],
+    in("r12") frames[12],
+    in("r13") frames[13],
+    in("r14") frames[14],
+    in("r15") frames[15],
+        options(nomem, nostack)
+    );
+
     // Extra hlt loop, in case hell freezes
```
Comment (lkatalin) [inline internal/shim-sev/src/asm.rs:71]: Not part of this PR, but this is a good comment : )

**Review by lkatalin — APPROVED** [review]
This is really useful! I'm admittedly a little fuzzy on some of what's happening in SEV here, but it all looks good as far as I can tell. Thank you also for the detailed commit messages and documentation! I will have to try this out.

File: internal/shim-sev/src/syscall.rs [hunk 2]
```diff
@@ -62,7 +65,6 @@ pub unsafe fn _syscall_enter() -> ! {
     push   rdi
     push   rsi
     push   rdx
-    push   rcx
```
[the remark this answers is not in the document]
Comment (haraldh) [reply internal/shim-sev/src/syscall.rs:65]: You are totally correct!
See https://github.com/enarx/enarx-keepldr/commit/67a22fc51154ba539e9d27f2f003359f7c42cedf
I shall make a big fat comment to prevent removal again, only because I think it is clobbered anyway.

[the remark this answers is not in the document]
Comment (haraldh) [reply internal/shim-sev/src/syscall.rs:65]: Fixed with https://github.com/enarx/enarx-keepldr/pull/166
